Documentation/RepoCompose.md

```diff
@@ -0,0 +1,289 @@
+
+Going forward repos will be given contracts, or APIs, that provide structured data about their dependencies and outputs.  This will give us the necessary data to compose our repos in our growing set of scenarios:
+
+- Linux distro builds from source
+- Official Windows signed build
+- Builds to quickly test and deploy new packages across repos
+
+These contracts will be included in the `run` command [specification](https://github.com/dotnet/buildtools/blob/master/Documentation/RunCommand.md).  This document in particular will be discussing the commands which allow us to compose the outputs of our repos.
+
+## Commands
+
+This document describes the commands used to compose different repos together.  There is a larger set of commands including `build`, `sync` and `clean` which is [described separately](https://github.com/dotnet/buildtools/blob/master/Documentation/Dev-workflow.md). 
+
+### consumes
+
+The `consumes` command returns json output that describes all of the external artifacts consumed by this repo.  This includes NuGet feeds, packages and arbitrary files from the web or file system.  The format of all these items is describe in the Artifact Specification section below. 
+
+The artifacts are grouped into sections: 
+
+- Build Dependencies: artifacts which are referenced by the build output and include NuGet packages, MSI, etc ...  In order to support a composed build these are further divided into floating and static dependencies:
+    - Floating: dependencies where versions can change as a part of a composed build via the `change` command.  This commonly includes CoreFx, CoreClr, etc ...  
+    - Static: dependencies whose version do not change via the `change` command.  This is commonly used for SDK binaries, legacy tools, etc ...  
+- Toolset Dependencies: artifacts used in the production of build output.  This commonly includes NuGet.exe, compilers, etc ... 
+
+These sections are identified respectively by the following JSON sections:
+
+``` json
+"dependencies": {
+    "floating": { 
+        // Build artifacts
+    },
+    "static": { 
+        // Static artifacts
+    },
+    "toolset": { 
+        // Toolset artifacts
+    }
+}
+```
+
+The data in the output is further grouped by operating system.  Builds of the same repo on different operating system can reasonably consume a different set of resources.  The output of `consumes` reflect this and allows per operating system dependencies:
+
+``` json
+{
+    "os-windows": {
+        "dependencies": { } 
+    },
+    "os-linux": {
+        "dependencies": { }
+    }
+}
+```
+
+In the case the `consumes` output doesn't want to take advantage of OS specific dependencies it can specify `"os-all"` as a catch all. 
+
+In addition to artifacts the consume feed can also optionally list any machine prerequitsites needed to build or test the repo:
```

Comment:
I understand this is not a simple problem to solve (implement) for all cases, but do we have any mechanism whereby we can also check for these prereq's or enable to repo to plugin their own prereq checking on a given machine?  I'd assume the answer is no, but I'd be pleasantly surprised if it wasn't or if something was in the works.  This may be a question more targeted at @maririos .